src/lib/features/playground/playground-service.ts

```diff
@@ -124,7 +124,20 @@ export class PlaygroundService {
                 this.resolveFeatures(filteredProjects, env),
             ),
         );
-        const contexts = generateObjectCombinations(context);
+
+        const { appName, properties, ...otherContextFields } = context;
+
+        const cleanedContextFields = Object.fromEntries(
```

Comment:
can we extract a function with a descriptive name and test it for all the corner cases so that we don't need to test for corner cases in the e2e test. Rationale: E2E tests are pretty slow and I'd leave them for the happy path.

Reply:
Done! I've left the e2e test because it still tests two important things:
1. We actually perform the cleaning of the context
2. We still return as `input` **exactly** what you passed in.

Extracted the function into its own file, added tests that check the edge cases I could think of.